src/main/java/seedu/address/model/person/PersonTagsContainsTagsPredicate.java

```diff
@@ -0,0 +1,33 @@
+package seedu.address.model.person;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.model.tag.Tag;
+
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches all of the tags given.
+ */
+public class PersonTagsContainsTagsPredicate implements Predicate<Person> {
+    private final List<Tag> tagList;
+
+    public PersonTagsContainsTagsPredicate(List<Tag> tagList) {
+        this.tagList = tagList;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return tagList.stream()
+                .allMatch(tag -> Arrays.stream(person.getTags().toArray()).anyMatch(personTag->personTag.equals(tag)));
```

Comment:
I'm struggling to follow this line, can you break this up into a more readable form/indentation